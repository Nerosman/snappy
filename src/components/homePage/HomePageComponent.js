import React from "react";
import {Jumbotron, Button} from "reactstrap"
import {Link} from "react-router-dom";

export const HomePageComponent = () => {
    return <Jumbotron>
        <h1 className="display-3">Welcome!</h1>
        <p className="lead">This app prepare some gift for you, click on the button below.</p>
        <hr className="my-2" />
        <p className="lead">
            <Link to="/ship-info"><Button>Get gift</Button></Link>
        </p>
    </Jumbotron>
}