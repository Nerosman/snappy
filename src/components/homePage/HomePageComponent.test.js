import React from "react";
import {configure, render, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {HomePageComponent} from "./HomePageComponent";

import {BrowserRouter as Router, Link} from 'react-router-dom';
import * as ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import {Button} from "reactstrap";

configure({adapter: new Adapter()});

describe('<HomePageComponent />', () => {
    it('renders without crashing', () => {
        shallow(<Router><HomePageComponent/></Router>);
    });
    it('can render div into HomePageComponent', () => {
        const div = document.createElement('div');
        render(<Router><HomePageComponent/></Router>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('check if component has one Jumbotron', () => {
        const wrapper = shallow(<HomePageComponent/>);
        expect(wrapper.find('Jumbotron').length).toBe(1);
    });
    test('snapshot homePage', () => {
        const wrapper = shallow(
            <Router><HomePageComponent/></Router>
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('renders elems', () => {
        const wrapper = render(<Router><HomePageComponent/></Router>);
        expect(wrapper.find('.display-3')).toHaveLength(1);
        expect(wrapper.find('.lead')).toHaveLength(2);
        expect(wrapper.find(".my-2")).toHaveLength(1);
    });

    test('Link matches snapshot', () => {
        const component = renderer.create(
            <Router>
                <HomePageComponent>
                    <Link to="/ship-info">
                        <Button>Get gift</Button>
                    </Link>
                </HomePageComponent>
            </Router>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});