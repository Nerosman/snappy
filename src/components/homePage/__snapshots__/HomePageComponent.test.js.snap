// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePageComponent /> Link matches snapshot 1`] = `
<div
  className="jumbotron"
>
  <h1
    className="display-3"
  >
    Welcome!
  </h1>
  <p
    className="lead"
  >
    This app prepare some gift for you, click on the button below.
  </p>
  <hr
    className="my-2"
  />
  <p
    className="lead"
  >
    <a
      href="/ship-info"
      onClick={[Function]}
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Get gift
      </button>
    </a>
  </p>
</div>
`;

exports[`<HomePageComponent /> snapshot homePage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserRouter>
    <HomePageComponent />
  </BrowserRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <HomePageComponent />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <HomePageComponent />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
